geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/kryo/KryoSimpleFeature.scala

```diff
@@ -0,0 +1,132 @@
+ */
+class KryoSimpleFeature(initialId: String, sft: SimpleFeatureType, initialValues: Array[AnyRef] = null)
+    extends SimpleFeature {
+
+  val featureId = new FeatureIdImpl(initialId)
+  val values = if (initialValues == null) Array.ofDim[AnyRef](sft.getAttributeCount) else initialValues
+
+  lazy private[this] val userData  = collection.mutable.HashMap.empty[AnyRef, AnyRef]
+  lazy private[this] val geomDesc  = sft.getGeometryDescriptor
+  lazy private[this] val geomIndex = if (geomDesc == null) -1 else sft.indexOf(geomDesc.getLocalName)
+
+  override def getFeatureType = sft
+  override def getType = sft
+  override def getIdentifier = featureId
+  override def getID = featureId.getID // this needs to reference the featureId, as it can be updated
+  override def getName: Name = sft.getName
+  override def getUserData = userData
+
+  override def getAttribute(name: Name) = getAttribute(name.getLocalPart)
+  override def getAttribute(name: String) = {
+    val index = sft.indexOf(name)
+    if (index == -1) null else getAttribute(index)
+  }
+  override def getAttribute(index: Int) = values(index)
+
+  override def setAttribute(name: Name, value: Object) = setAttribute(name.getLocalPart, value)
+  override def setAttribute(name: String, value: Object) = {
+    val index = sft.indexOf(name)
+    if (index == -1) {
+      throw new IllegalArgumentException(s"Attribute $name does not exist in type $sft")
+    }
+    setAttribute(index, value)
+  }
+  override def setAttribute(index: Int, value: Object) = {
+    val binding = sft.getDescriptor(index).getType.getBinding
+    values(index) = Converters.convert(value, binding).asInstanceOf[AnyRef]
+  }
+
+  // following methods delegate to setAttribute to get type conversion
+  override def setAttributes(vals: JList[Object]) =
+    vals.zipWithIndex.foreach { case (v, i) => setAttribute(i, v) }
+  override def setAttributes(vals: Array[Object]) =
+    vals.zipWithIndex.foreach { case (v, i) => setAttribute(i, v) }
+
+  override def getAttributeCount = values.length
+  override def getAttributes: JList[Object] = values.toList
+
+  override def getDefaultGeometry: Object = if (geomIndex == -1) null else getAttribute(geomIndex)
+  override def setDefaultGeometry(geo: Object) = setAttribute(geomIndex, geo)
+
+  override def getBounds: BoundingBox = getDefaultGeometry match {
+    case g: Geometry => new ReferencedEnvelope(g.getEnvelopeInternal, sft.getCoordinateReferenceSystem)
+    case _ => new ReferencedEnvelope(sft.getCoordinateReferenceSystem)
+  }
+
+  override def getDefaultGeometryProperty =
+    if (geomDesc == null) null else new GeometryAttributeImpl(getDefaultGeometry, geomDesc, null)
+  override def setDefaultGeometryProperty(geoAttr: GeometryAttribute) =
+    if (geoAttr == null) setDefaultGeometry(null) else setDefaultGeometry(geoAttr.getValue)
+
+  override def getProperties: JCollection[Property] =
+    getAttributes.zip(sft.getAttributeDescriptors).map {
```

Comment:
I think I might rename this class to just 'ScalaSimpleFeature', since it's not really kryo-specific at all, except for the fact that it's efficient to instantiate it from kryo.